### PR TITLE
#2 Change commands to be independent of database

### PR DESCRIPTION
**Pull request by KamilWojciech**

### CODE REVIEW

**Comment by kedrap** [comment]
Change file name Command/SchedulerJobAddCommand.php => Command/JobInsertCommand.php, etc

File: Resources/config/services.yml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+services:
+    svd_scheduler.job_provider.odm:
+        calls:
+            -                           [setDocumentManager, [@doctrine.odm.mongodb.document_manager]]
+        class:                          Svd\SchedulerBundle\Provider\Job\Odm
```
Comment (duzymaju) [inline Resources/config/services.yml]: No new line.

**Comment by duzymaju** [comment]
Maybe job providers should have method names like "get", "getOne", "update", "delete" and so on instead of "getJobs", "getJob", "updateJob", "deleteJob" etc.
If not, :+1:.

**Comment by KamilWojciech** [comment]
Me and @kedrap talked about that, and we decided to use Job on method names, because it is job manager not job object and you can have methods not strictly connected to job object

**Comment by duzymaju** [comment]
:+1:

File: Command/JobAddCommand.php [hunk 2]
```diff
@@ -49,19 +48,11 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
         CronExpression::factory($expression);
```
Comment (kedrap) [inline Command/JobAddCommand.php]: If you use this construction to check if $expression is valid, maybe better will be to use isValidExpression($expression) function?

File: Command/JobAddCommand.php [hunk 3]
```diff
@@ -26,11 +26,10 @@ protected function configure()
     {
         parent::configure();
 
-        $this->setName('scheduler:job:add')
+        $this->setName('job:add')
             ->setDescription('Add job to scheduler')
             ->addArgument('expression', InputArgument::REQUIRED, 'Cron expression')
             ->addArgument('cmd', InputArgument::REQUIRED, 'Command to run')
```
Comment (kedrap) [inline Command/JobAddCommand.php]: Let's change it from 'cmd' to 'command', please :)

Comment (KamilWojciech) [inline Command/JobAddCommand.php]: You can't use command as argument name
```
[LogicException]
An argument with name "command" already exists.
```

File: DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -18,7 +18,12 @@ class Configuration implements ConfigurationInterface
...
-        $treeBuilder->root('svd_scheduler');
+        $rootNode = $treeBuilder->root('svd_scheduler');
+
+        $rootNode
+            ->children()
+                ->scalarNode('job_provider')->end()
+            ->end();
 
         return $treeBuilder;
     }
```
Comment (kedrap) [inline DependencyInjection/Configuration.php]: Remember to add newline after line #30 :)

File: Provider/Job/Odm.php [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    {
+        $this->dm = $documentManager;
+
+        return $this;
+    }
+
+    /**
+     * Get jobs
+     *
+     * @param array $criteria
```
Comment (kedrap) [inline Provider/Job/Odm.php]: Fix it from:
@param array $criteria
to:
@param array $criteria criteria

File: Provider/Job/Odm.php [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+        $jobRepository = $this->dm
+            ->getRepository('SvdSchedulerBundle:Job');
+
+        return $jobRepository->findBy($criteria);
+    }
+
+    /**
+     * Get job
+     *
+     * @param array $criteria
```
Comment (kedrap) [inline Provider/Job/Odm.php]: Fix also

File: Provider/Job/Odm.php [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+        if ($flush) {
+            $this->dm
+                ->flush();
+        }
+
+        return $job;
+    }
+
+    /**
+     * @param      $expression
```
Comment (kedrap) [inline Provider/Job/Odm.php]: Fix also and add parameter type

File: Provider/Job/Odm.php [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+     * @param      $command
+     * @param null $status
```
Comment (kedrap) [inline Provider/Job/Odm.php]: Does this $status parameter has really null type? ;)

Or shouldn't it be:
@param integer $status status?

**Comment by kedrap** [comment]
:+1:

**Pull request merged**